### ISSUE TITLE
When using native scrolling anchor pane height gives wrong values.

### ISSUE THREAD

**Issue by guptamitesh** [JPro-one/JPro-tickets#70]
Hi,
I modified the sample https://github.com/JPro-one/JPro-Samples/blob/master/native-scrolling and replaced the code inside NativeScrollingSample.java with :
```
StackPane rootStackPane;

    @Override
    public void start(Stage primaryStage) throws Exception {
        rootStackPane = new StackPane();

        AnchorPane anchorPane = new AnchorPane();
        VBox container = new VBox() {
            @Override
            protected void layoutChildren() {
                if ((this.getScene() != null) && WebAPI.isBrowser()) {
                    getWebAPI().requestLayout(this.getScene());
                    super.layoutChildren();
                } else {
                    super.layoutChildren();
                }
            }
        };

        VBox container2 = new VBox();

        Button button = new Button("Add message");
        button.onActionProperty().set((e) -> {
            addField(container2);
        });

        container.getChildren().add(button);
        container.getChildren().add(container2);


        container.setBackground(new Background(new BackgroundFill(Color.rgb(255, 0, 0, 1), CornerRadii.EMPTY, Insets.EMPTY)));

        anchorPane.getChildren().add(container);
        AnchorPane.setTopAnchor(container, 0.0);
        AnchorPane.setLeftAnchor(container, 0.0);
        AnchorPane.setBottomAnchor(container, 0.0);
        AnchorPane.setRightAnchor(container, 0.0);
        anchorPane.setBackground(new Background(new BackgroundFill(Color.rgb(0, 0, 255, 1), CornerRadii.EMPTY, Insets.EMPTY)));

        rootStackPane.getChildren().add(anchorPane);
        Scene scene = new Scene(rootStackPane);
        scene.getStylesheets().add("/css/Kiwi.css");
        scene.getStylesheets().add("/css/BZColorpicker.css");
        primaryStage.setScene(scene);
        primaryStage.show();
    }

    public void addField(VBox container) {
        FlowPane flowPane = new FlowPane();
        flowPane.setOrientation(Orientation.HORIZONTAL);
        flowPane.setHgap(38);
        flowPane.setVgap(38);
        flowPane.setAlignment(Pos.CENTER_LEFT);
        container.getChildren().add(flowPane);


        for(int i=0; i<5; i++) {
            Pane pane = new Pane();
            pane.setMinSize(212, 212);
            pane.setBackground(new Background(new BackgroundFill(Color.rgb(0, 255, 0, 1), CornerRadii.EMPTY, Insets.EMPTY)));
            pane.setPadding(new Insets(0,0,0,23));
            flowPane.getChildren().add(pane);
        }
    };
```
On pressing the addMessage button it can be observed that scroll height(Red Area) is coming more than the actual container height. (Green area). If we remove the anchor pane and the container object directly to rootStackPane the code works as expected. 

Attached image for reference when using anchor pane:
![image](https://user-images.githubusercontent.com/13232814/89683823-de482780-d916-11ea-85f8-e23f8a75b085.png)

Without Anchor Pane:
![image](https://user-images.githubusercontent.com/13232814/89683988-29fad100-d917-11ea-83b2-ab91e3d5cb87.png)

Hope the above code will give you the repro needed. Try increasing the for loop count for(int i=0; i<5; i++) in case the containers are not filling up the entire width of your screen.

Jpro embedded with following options:
```
<jpro-app href="localhost:8080/app/default" fxheight="true" nativeScrolling="true" fullscreen="true"/>
```

Thanks,
Mitesh

**Comment by FlorianKirmaier** [comment]
Thank you for the bug report! 
After investigating it, I have to say this is probably not a bug in JPro but a bug in the AnchorPane.
If you use normal JavaFX and put the AnchorPane into a ScrollPane, you get similar weird behavior.
Probably, it would be fixable in JavaFX with some time investment.

I would recommend avoiding AnchorPane when layouting is important and the target width and height are unknown.

Small note:
```
            @Override
            protected void layoutChildren() {
                if ((this.getScene() != null) && WebAPI.isBrowser()) {
                    getWebAPI().requestLayout(this.getScene());
                    super.layoutChildren();
                } else {
                    super.layoutChildren();
                }
            }
```
The method should be overwritten for the root node, not for some other node.

**Comment by guptamitesh** [comment]
Thank you for the reply @FlorianKirmaier . One more problem I am facing is making inner components native scrollable.
One quick example for same:
```
    @Override
    public void start(Stage primaryStage) {
        Rectangle2D screenBounds = getWebAPI().getBrowserSize();
        rootStackPane = new StackPane() {
            @Override
            protected void layoutChildren() {
                if ((this.getScene() != null) && WebAPI.isBrowser()) {
                    getWebAPI().requestLayout(this.getScene());
                    super.layoutChildren();
                } else {
                    super.layoutChildren();
                }
            }
        };

        VBox container = new VBox();
        container.setMinHeight(screenBounds.getHeight());
        container.setMaxHeight(screenBounds.getHeight());
        container.setPrefHeight(screenBounds.getHeight());

        VBox container2 = new VBox();

        container2.setSpacing(20);
        container2.setBackground(new Background(new BackgroundFill(Color.rgb(255, 255, 0, 1), CornerRadii.EMPTY, Insets.EMPTY)));
        Button button = new Button("Add message");
        button.onActionProperty().set((e) -> {
            addField(container2);
        });

        container.getChildren().add(button);
        container.setBackground(new Background(new BackgroundFill(Color.rgb(255, 0, 0, 1), CornerRadii.EMPTY, Insets.EMPTY)));

        StackPane stackPane = new StackPane();
        stackPane.getChildren().add(container2);
        container.getChildren().add(stackPane);
        //ScrollPane scrollPane = new ScrollPane(container2);
        //scrollPane.setFitToWidth(true);
        //scrollPane.setFitToHeight(true);
        //container.getChildren().add(scrollPane);

        rootStackPane.getChildren().add(container);
        Scene scene = new Scene(rootStackPane);
        primaryStage.setScene(scene);
        primaryStage.show();
    }
    int count = 1;
    public void addField(VBox container) {
        FlowPane flowPane = new FlowPane();
        flowPane.setOrientation(Orientation.HORIZONTAL);
        flowPane.setHgap(38);
        flowPane.setVgap(38);
        flowPane.setAlignment(Pos.CENTER_LEFT);
        container.getChildren().add(flowPane);


        for(int i=0; i<5; i++) {
            Pane pane = getPane();
            flowPane.getChildren().add(pane);
        }
    }

    public Pane getPane() {
        Pane pane = new Pane();
        Label label = new Label("Count: " +count);
        pane.getChildren().add(label);
        pane.setMinSize(212, 212);
        pane.setBackground(new Background(new BackgroundFill(Color.rgb(0, 255, 0, 1), CornerRadii.EMPTY, Insets.EMPTY)));
        pane.setPadding(new Insets(0,0,0,23));
        count++;
        return pane;
    }
```
Here i want container2 to be scrollable with native browser scroll, not able to achieve it.
If I use ScrollPane instead on StackPane in the code above it is the behaviour I am looking for (code for scrollpane is commented). Using scrollpane here kills the purpose as it is not using native browser scroll and is making my whole application looking laggy when scrolled. 

Basically want to scroll inner div independently without disturbing other page components.

Any help in how to achieve this?

Thanks,
Mitesh

**Comment by FlorianKirmaier** [comment]
The nativeScrolling basically makes JavaFX determine the size of whole application.
Except that, it doesn't change the behavior of JavaFX.

```
        container.setMinHeight(screenBounds.getHeight());
        container.setMaxHeight(screenBounds.getHeight());
        container.setPrefHeight(screenBounds.getHeight());
```
If you want to fix the size to the height of the browser, you can just ignore the nativeScrolling sample, it's the default behavior for JPro.
When you only set the minHeight, you will get native scrolling but always filling at least the whole page.

You can compare the lag of ScrollPanes with one of our samples. If its faster in our samples, then probably for some reason your server has some latency. In this case, you should check why, because it will probably negatively affect the user experience also in other cases.

Making inner components Native Scrollable,
would be a bit more complicated.
Now, with the new div-based rendering engine (compared to the old SVG based) 
it should be possible to implement that properly.
But that would be a new JPro Feature.
It's probably even possible to implement it as a skin for ScrollPane.
In this case, it would be possible to enable it later through CSS.

**Comment by guptamitesh** [comment]
Thanks for comments @FlorianKirmaier . I believe it would be a useful and powerful feature to have. Would love to see it in some future release. Please keep us updated if feature comes into plan in near future.

Thanks,
Mitesh
